drivers/sensor/ite_tach_it8xxx2/tach_ite_it8xxx2.c

```diff
@@ -0,0 +1,282 @@
+ *                            |       capture pulses                  T0B (GPJ2)
+ *                            |       in sample rate
+ *                            |       period
+ *           +-----------+    |
+ * Crystal-->| Prescaler |--->|        Tachometer 1                   T1A (GPD7)
+ * 32.768k   +-----------+    |             |          +-----------+   |
+ *                            |       +-----+-----+    |   _   _   |<--+
+ *                            |------>| F2TL/MRR  |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T1B (GPJ3)
+ *                            |       in one second
+ *                            |       period
+ *                            |
+ *
+ * Based on the counter value, we can compute the current RPM of external signal
+ * from encoders.
+ */
+
+#include <device.h>
+#include <drivers/pinmux.h>
+#include <drivers/sensor.h>
+#include <dt-bindings/sensor/it8xxx2_tach.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tach_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * NOTE: The PWM output maximum is 324Hz in EC LPM, so if we need fan to work
+ *       then don't let EC enter LPM.
+ */
+#define TACH_FREQ		EC_FREQ
+
+/* Device config */
+enum tach0_pinctrl_idx {
+	TACH0A = 0,
+	TACH0B,
+};
+
+enum tach1_pinctrl_idx {
+	TACH1A = 0,
+	TACH1B,
+};
+
+struct tach_alt_cfg {
+	/* Pinmux control device structure */
+	const struct device *pinctrls;
+	/* GPIO pin */
+	uint8_t pin;
+	/* Alternate function */
+	uint8_t alt_fun;
+};
+
+struct tach_it8xxx2_config {
+	/* Fan x tachometer LSB reading register */
+	uintptr_t reg_fxtlrr;
+	/* Fan x tachometer MSB reading register */
+	uintptr_t reg_fxtmrr;
+	/* Tachometer switch control register */
+	uintptr_t reg_tswctlr;
+	/* Tachometer alternate configuration list */
+	const struct tach_alt_cfg *alt_list;
+	/* Select channel of tachometer */
+	int channel;
+	/* Number of pulses per round of tachometer's input */
+	int pulses_per_round;
+};
+
+/* Driver data */
+struct tach_it8xxx2_data {
+	/* Captured counts of tachometer */
+	uint32_t capture;
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev)	((const struct tach_it8xxx2_config *)(dev)->config)
+#define DRV_DATA(dev)	((struct tach_it8xxx2_data *)(dev)->data)
+
+static int tach_ch_valid(const struct device *dev, int tach_ch)
+{
+	const struct tach_it8xxx2_config *const config = DRV_CONFIG(dev);
+	volatile uint8_t *reg_tswctlr = (uint8_t *)config->reg_tswctlr;
+	int valid = 0;
+
+	switch (tach_ch) {
+	case TACH0_CHANNEL_A:
+		if ((*reg_tswctlr & 0x0C) == IT8XXX2_PWM_T0DVS)
+			valid = 1;
+		break;
+	case TACH0_CHANNEL_B:
+		if ((*reg_tswctlr & 0x0C) == (IT8XXX2_PWM_T0DVS | IT8XXX2_PWM_T0CHSEL))
+			valid = 1;
+		break;
+	case TACH1_CHANNEL_A:
+		if ((*reg_tswctlr & 0x03) == IT8XXX2_PWM_T1DVS)
```

Comment:
Can you create an IT8XXX2_PWM_xxxxx_MASK to remove the 0x03 magic number?

Reply:
Done